### PR TITLE
chore: [C4] Add first batch of QA fixes 

### PR DESCRIPTION
**Pull request by YamenMerhi**
# What does this PR introduce?

Add some of the QA fixes 

### PR Checklist

- [x] Wrote Tests
- [x] Wrote & Generated Documentation (readme/natspec/dodoc)
- [x] Ran `npm run lint` && `npm run lint:solidity` (solhint)
- [x] Ran `npm run format` (prettier)
- [x] Ran `npm run build`
- [x] Ran `npm run test`

### CODE REVIEW

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 1]
```diff
@@ -203,16 +204,16 @@ abstract contract LSP0ERC725AccountCore is
             emit ValueReceived(msg.sender, msg.value);
         }
 
-        address _owner = owner();
+        address accountOwner = owner();
```
Comment (CJ42) [inline contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:207]: nice 👌  variable shadowing here?

File: contracts/LSP6KeyManager/LSP6KeyManagerCore.sol [hunk 2]
```diff
@@ -305,13 +305,8 @@ abstract contract LSP6KeyManagerCore is
...
         bytes calldata data
     ) external virtual returns (bytes4) {
-        bool isSetData = false;
-        if (
-            bytes4(data) == IERC725Y.setData.selector ||
-            bytes4(data) == IERC725Y.setDataBatch.selector
-        ) {
-            isSetData = true;
-        }
+        bool isSetData = bytes4(data) == IERC725Y.setData.selector ||
```
Comment (CJ42) [inline contracts/LSP6KeyManager/LSP6KeyManagerCore.sol:308]: great 👌

File: tests/UniversalProfile.behaviour.ts [hunk 3]
```diff
@@ -133,6 +133,15 @@ export const shouldBehaveLikeLSP3 = (
       '0xdaea594e385fc724449e3118b2db7e86dfba1826',
     ];
 
+    it('should fail when setting empty data keys', async () => {
```
Comment (CJ42) [inline tests/UniversalProfile.behaviour.ts:136]:
```suggestion
    it('should fail when passing empty arrays of data keys / values', async () => {
```

File: contracts/LSP20CallVerification/LSP20CallVerification.sol [hunk 4]
```diff
@@ -39,7 +39,7 @@ abstract contract LSP20CallVerification {
         if (bytes3(magicValue) != bytes3(ILSP20.lsp20VerifyCall.selector))
             revert LSP20InvalidMagicValue(false, returnedData);
 
-        return magicValue[3] == 0x01 ? true : false;
+        return bytes1(magicValue[3]) == 0x01;
```
Comment (CJ42) [inline contracts/LSP20CallVerification/LSP20CallVerification.sol:42]: Type casting not needed here. Index access on `bytesN` returns a `bytes1` already.

```suggestion
        return magicValue[3] == 0x01;
```

**Review by CJ42 — CHANGES_REQUESTED** [review]

File: contracts/LSP20CallVerification/LSP20CallVerification.sol (hunk 4 above)
Comment (YamenMerhi) [reply contracts/LSP20CallVerification/LSP20CallVerification.sol:42]: Being done in another PR

**Review by b00ste — APPROVED** [review]

File: contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol [hunk 5]
```diff
@@ -584,24 +592,24 @@ abstract contract LSP0ERC725AccountCore is
      *
      * @custom:requirements Can be only called by the {owner} or by an authorised address that pass the verification check performed on the owner.
      *
-     * @custom:danger Leaves the contract without an owner. Once ownership of the contract has been renounced, any functions that are restricted to be called by the owner will be permanently inaccessible, making these functions not callable anymore and unusable.
+     * @custom:danger Leaves the contract without an owner. Once ownership of the contract has been renounced, any functions that are restricted to be called by the owner or an address allowed by the owner will be permanently inaccessible, making these functions not callable anymore and unusable.
```
[the remark this answers is not in the document]
Comment (YamenMerhi) [reply contracts/LSP0ERC725Account/LSP0ERC725AccountCore.sol:586]: LSP9 have no LSP20 so this does not apply

**Review by CJ42 — COMMENTED** [review]
added some small review comments

**Review by CJ42 — APPROVED** [review]

**Pull request merged**